packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js

```diff
@@ -0,0 +1,34 @@
+const { env: { CLIENT_URL } } = process
+
+export default class NotificationScannerBaseModel {
+  async shouldNotifyUser(user) {
+    user = await user.$fetchGraph('[email, notificationSetting]')
+    return (
+      user?.email &&
+      !user?.notificationSetting?.notificationsDisabled &&
+      (user?.emailVerified || this._sendUnverified)
+    )
+  }
+  get _MINUTES() { return 60 * 1000 }
+  get _HOURS() { return 60 * this._MINUTES }
+  get _DAYS() { return 24 * this._HOURS }
+  get _CLIENT_URL() { return CLIENT_URL }
+
+  // mandatory methods and functions for extended models
```

Comment:
In terms of being mandatory, as JS doesn't have a compile-time step for validating abstract classes, you may want to throw an error in the base implementation instead of returning a null default.

Reply:
created issue #198